cmd/mapstructure-to-hcl2/mapstructure-to-hcl2.go

```diff
@@ -222,67 +222,76 @@ func outputHCL2SpecField(w io.Writer, accessor string, fieldType types.Type, tag
 		fmt.Fprintf(w, `(&%s{}).HCL2Spec()`, fieldType.String())
 		return
 	}
+	spec, specType := writeSpecField(accessor, fieldType)
+	switch specType {
+	case cty.NilType:
+		fmt.Fprintf(w, spec.(string))
+	default:
+		fmt.Fprintf(w, `%#v`, spec)
+	}
+
+}
+
+func writeSpecField(accessor string, fieldType types.Type) (interface{}, cty.Type) {
```

Comment:
Nicely done! I can understand how this is working, but I have a hard time mapping the types back to their configuration schema without looking at the generated code. It would help to provide some comments here and maybe some external documentation, which I think  you are already working on, to explain the specType mapping a little. For example `types.Named` can represent a nested structure and what else?

Reply:
That's a good idea! I will try adding some comments with @azr's help because there's some code I still need to check the generated code as you did,

Reply:
About the comments, yes we need more of those ! We can also do them in a pairing session they we probably come out super good.